projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts

```diff
@@ -222,9 +256,29 @@ export class IgxToggleDirective implements IToggleView, OnInit, OnDestroy {
      * ```
      */
     public close() {
+        if (this._collapsed) {
```

Comment:
Add check for closing animation in progress. If such animation is in progress return too.

Reply:
Still toggle method into igxToggleAction is not relying on the correct flags. Needs conjunction between collapsed state and animation events both opening and closing.